R/plots.R

```diff
@@ -268,25 +311,17 @@ getPlottingData <- function(study, modelID, featureID, testID = NULL, libraries
   featureID <- unique(featureID)
 
   if (length(modelID) > 1) {
-    mapping <- getMapping(study, libraries = libraries)
-    listMaxLength <- max(sapply(mapping, length))
-    mapping <- lapply(lapply(mapping, unlist), "length<-", listMaxLength)
-
-    mappingdf <- as.data.frame(mapping, stringsAsFactors = FALSE)
-    mapping_features_all <- stats::na.omit(mappingdf)
-    column_order <- unique(c(modelID[1], colnames(mapping_features_all)))
-    mapping_features_all <- mapping_features_all[, column_order]
-
-    mapping_features <- mapping_features_all[which(mapping_features_all[,1] %in% featureID),]
-
-    testID_all <- testID
+    mappingPlottingData <- getMappingPlottingData(study, modelID, featureID, testID, libraries)
+    list2env(mappingPlottingData, environment())
```

Comment:
Please don't use `list2env()`. This makes it very hard to reason about. `R CMD check` threw the following NOTE:

```
* checking R code for possible problems ... NOTE
getPlottingData: no visible binding for global variable
  'mapping_features'
getPlottingData: no visible binding for global variable 'testID_all'
Undefined global functions or variables:
  mapping_features testID_all
```

I think `mapping_features` and `testID_all` are created by the call to `list2env()`. To make it easier to follow the code, please explicitly assign these variables:

```R
mapping_features <- mappingPlottingData$mapping_features
testID_all <- mappingPlottingData$testID_all
modelID <- mappingPlottingData$modelID
```

This also alerts the reader to the fact that `modelID` has been modified (looks like it is sorted in `getMappingPlottingData`)

Reply:
Updated. List2env is replaced by explicit variable attribution, as indicated.